Much better Pro docs (#4263)
zenml-io/zenml: docs/book/getting-started/zenml-pro/saas-deployment.md

- **POC and pilot projects** that need to demonstrate value quickly

## Secret Management Options

### Default: ZenML-Managed Secrets Store

By default, ZenML Pro SaaS stores your cloud credentials securely in our managed secrets store. This provides:
- Zero configuration required
- Automatic encryption at rest and in transit
- Access controls via RBAC
- Audit logging of secret access

### Alternative: Customer-Managed Secrets Store

For organizations with strict security requirements, you can configure ZenML to use your own secrets management solution:
- AWS Secrets Manager
- Google Cloud Secret Manager
- Azure Key Vault
- HashiCorp Vault

![SaaS with customer secret store](../../.gitbook/assets/cloud_architecture_saas_detailed_2.png)

This keeps all credentials within your infrastructure while still benefiting from managed ZenML services - [Book a call](https://www.zenml.io/book-your-demo) with us if you want this set up.

## Network Architecture

### Outbound-Only Communication

ZenML Pro SaaS uses outbound-only connections from your infrastructure to ZenML services:
- No inbound connections required to your infrastructure
- Compatible with firewall and VPN restrictions

> - Secure WebSocket connections for real-time updates

Reviewer: This is not entirely accurate:

* inbound connections are required for some features:
  * showing logs in the UI
  * showing artifact visualizations in the UI
  * triggering pipelines from the UI
  * secret management when a custom secrets stores is configured
* definitely NOT compatible with firewall and VPN restrictions (in fact, this is a good point why someone might choose the hybrid option over the full SaaS option)
* what is this websocket connection feature? I don't remember us implementing something like that

### Artifact Store Access

> The ZenML dashboard requires read access to your artifact store to display:

Reviewer: This should say UI (not dashboard) and it requires access to more than the artifact store (log store, orchestrators; see my other comment)

- Pipeline visualizations
- Model comparison views
- Artifact lineage graphs
- Step logs and outputs

> You control this access by configuring appropriate cloud IAM permissions.

Reviewer: This is misleading. The truth is this: if you give your users permission to access these things, you also implicitly give the UI permission to do so. You could say "you control who can access this information in the UI by configuring appropriate ZenML Pro RBAC permissions.

Author: no don't we have some setting for our wrokspace/UI to enable/disable this?

- **Software Bill of Materials (SBOM)**: Complete dependency inventory for compliance
- **Compliance documentation**: Support for your security audits and certifications

Contact [cloud@zenml.io](mailto:cloud@zenml.io) to request security documentation.

## Pricing & Support

ZenML Pro SaaS includes:
- Managed infrastructure and updates
- Professional support with SLA
- Regular security patches and updates
- Access to pro-exclusive features
- Usage-based pricing model

[Contact us](https://www.zenml.io/book-your-demo) for pricing details and custom plans.

## Comparison with Other Deployments

| Feature | SaaS | Hybrid SaaS | Self-hosted |
|---------|------|-------------|------------|
| Setup Time | ⚡ Minutes | Hours | Days |
| Maintenance | Zero | Workspace only | Full stack |
| Infrastructure Control | Minimal | Moderate | Complete |
| Data Sovereignty | Metadata on ZenML | Full | Full |
| Best For | Fast time-to-value | Security requirements | Strictest compliance |

[Compare all deployment options →](README.md#deployment-scenarios)

## Migration Path

Already running ZenML OSS? Migrating to SaaS is straightforward:

> 1. **Export your data**: Use ZenML's migration tools

Reviewer: what tools would those be exactly ? I don't remember us implementing any migration tools.

2. **Set up SaaS workspace**: Configure your cloud connections
3. **Import metadata**: Transfer existing pipeline history

> 4. **Update pipelines**: Point to your new ZenML server

Reviewer: This entire section feels out of place. You only have to mention that it's possible to migrate an OSS server to SaaS/Pro while keeping the existing data and URL. The only thing that would have to change is how users are authenticating.

---
description: Learn about ZenML Pro SaaS deployment - the fastest way to get started with production-ready MLOps.
icon: cloud
---

# SaaS Deployment

ZenML Pro SaaS is the fastest and easiest way to get started with enterprise-grade MLOps. With zero infrastructure setup required, you can be running production pipelines within minutes while maintaining full control over your data and compute resources.

{% hint style="info" %}

> To try ZenML Pro SaaS or to learn more, [book a call](https://www.zenml.io/book-your-demo).

Reviewer 2: ```suggestion
To get access to ZenML Pro, [book a call](https://www.zenml.io/book-your-demo).
```

{% endhint %}

## Overview

In a SaaS deployment, ZenML manages all server infrastructure while your sensitive data and compute resources remain in your own cloud environment. This architecture provides the fastest time-to-value while maintaining data sovereignty for your ML workloads.

![ZenML Pro SaaS deployment architecture](../../.gitbook/assets/cloud_architecture_scenario_1.png)

## Architecture

### What Runs Where

| Component | Location | Purpose |
|-----------|----------|---------|

> | **ZenML Server** | ZenML Infrastructure | Manages pipeline orchestration and metadata |

Reviewer 2: as this isnt the OSS zenml server is it worth detailing what all these components are and saying this is a special fork of the zenml oss server?

- Access controls via RBAC
- Audit logging of secret access

### Alternative: Customer-Managed Secrets Store

For organizations with strict security requirements, you can configure ZenML to use your own secrets management solution:
- AWS Secrets Manager
- Google Cloud Secret Manager
- Azure Key Vault
- HashiCorp Vault

![SaaS with customer secret store](../../.gitbook/assets/cloud_architecture_saas_detailed_2.png)

This keeps all credentials within your infrastructure while still benefiting from managed ZenML services - [Book a call](https://www.zenml.io/book-your-demo) with us if you want this set up.

## Network Architecture

### Outbound-Only Communication

ZenML Pro SaaS uses outbound-only connections from your infrastructure to ZenML services:
- No inbound connections required to your infrastructure
- Compatible with firewall and VPN restrictions
- Secure WebSocket connections for real-time updates

### Artifact Store Access

The ZenML dashboard requires read access to your artifact store to display:
- Pipeline visualizations
- Model comparison views
- Artifact lineage graphs
- Step logs and outputs

You control this access by configuring appropriate cloud IAM permissions.

> ## Getting Started

Reviewer 2: IMO super strnage to have this whole section here...
